### PR TITLE
Delete `ProtocolUTXOCostPerWord` feature

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Delete `ProtocolUTXOCostPerWord` feature
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Delete `ProtocolUTXOCostPerWord` feature

This PR is currently not branched off tip of main so that I can build this https://github.com/input-output-hk/cardano-cli/pull/400

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]
LGTM! One small comment.

File: cardano-api/internal/Cardano/Api/Ledger/Lens.hs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  , invalidHereAfterL
+  ) where
+
+import           Cardano.Ledger.Allegra.Scripts
+import           Cardano.Ledger.BaseTypes (SlotNo, StrictMaybe (..))
+
+import           Lens.Micro
+
+-- Ideally this should be defined in cardano-ledger
+invalidBeforeL :: Lens' ValidityInterval (StrictMaybe SlotNo)
```
[the remark this answers is not in the document]
Comment (lehins) [reply cardano-api/internal/Cardano/Api/Ledger/Lens.hs:15]: Yes, we can definitely add this to the `cardano-ledger-api`. Feel free to submit a PR that adds them to this module an exports these lenses together with `ValidityInterval`:
https://github.com/input-output-hk/cardano-ledger/blob/8415ca55d91f3b4e50c10c80215a16ca707592e5/libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs#L34

[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Ledger/Lens.hs:15]: Okay will do!

[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/Ledger/Lens.hs]: https://github.com/input-output-hk/cardano-ledger/pull/3807

**Pull request merged**